### PR TITLE
bicep: init at 0.26.54

### LINKED ISSUE #282826: Package request: bicep
**Project description**

Bicep is a declarative language for describing and deploying Azure resources 

It can also be used as an extension of `azure-cli` tool.

**Metadata**

* homepage URL: https://github.com/Azure/bicep
* source URL: https://github.com/Azure/bicep
* license: mit
* platforms: unix, linux, darwin, windows
---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by khaneliman**
## Description of changes
Resolves https://github.com/NixOS/nixpkgs/issues/282826

Currently getting a compile error on darwin. Going to mark as broken on darwin and ship it as is so it is at least available to linux users while I try and figure out how to get past the darwin compilation issue.
`> /private/tmp/nix-build-bicep-0.24.24.drv-0/source/src/Bicep.Cli/Services/TemplateEvaluator.cs(29,9): error SYSLIB1042: The specified regex is invalid. 'cultureName' [/private/tmp/nix-build-bicep-0.24.24.drv-0/source/src/Bicep.Cli/Bicep.Cli.csproj]
       > /private/tmp/nix-build-bicep-0.24.24.drv-0/source/src/Bicep.Cli/Services/TemplateEvaluator.cs(30,38): error CS8795: Partial method 'TemplateEvaluator.templateSchemaPattern()' must have an implementation part because it has accessibility modifiers. [/private/tmp/nix-build-bicep-0.24.24.drv-0/source/src/Bicep.Cli/Bicep.Cli.csproj]
`

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/bi/bicep/package.nix [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+, nix-update-script
+}:
+
+buildDotnetModule rec {
+  pname = "bicep";
+  version = "0.26.54";
+
+  src = fetchFromGitHub {
+    owner = "Azure";
+    repo = pname;
```
Comment (drupol) [inline pkgs/by-name/bi/bicep/package.nix:16]: Please don't use this.

`repo = pname;` is nice for DRY but creates a binding that goes too far.

See further information about this here: https://github.com/nix-community/nixpkgs-lint/issues/21

File: pkgs/by-name/bi/bicep/package.nix [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+  doCheck = !(stdenv.isDarwin && stdenv.isAarch64); # mono is not available on aarch64-darwin
+
+  nativeCheckInputs = [ mono ];
+
+  testProjectFile = "src/Bicep.Cli.UnitTests/Bicep.Cli.UnitTests.csproj";
+
+  passthru.updateScript = ./updater.sh;
+
+  meta = with lib; {
```
Comment (drupol) [inline pkgs/by-name/bi/bicep/package.nix:37]: Could you remove with lib; ?

File: pkgs/by-name/bi/bicep/package.nix [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    broken = stdenv.isDarwin;
+    description = "Domain Specific Language (DSL) for deploying Azure resources declaratively";
+    homepage = "https://github.com/Azure/bicep/";
+    changelog = "https://github.com/Azure/bicep/releases/tag/v${version}";
+    license = licenses.mit;
+    maintainers = with maintainers; [ ];
```
Comment (drupol) [inline pkgs/by-name/bi/bicep/package.nix:43]: How about adding yourself in the maintainer list ?

**Review by drupol — CHANGES_REQUESTED** [review]

File: pkgs/by-name/bi/bicep/package.nix (hunk 1 above)
Comment (khaneliman) [reply pkgs/by-name/bi/bicep/package.nix]: Sorry, copied an existing dotnet package and forgot to clean it up after getting it to build.

**Review by drupol — APPROVED** [review]

**Comment by khaneliman** [comment]
@drupol looks like the checks finally cleared

**Pull request merged**